src/store/modules/stock/actions.ts

```diff
@@ -26,6 +28,55 @@ const actions: ActionTree<StockState, RootState> = {
       logger.error(err)
       showToast(translate('No data available!'))
     }
+  },
+
+  async fetchInventoryCount({ commit }, { productId }) {
+    try {
```

Comment:
Add a check that if the information for the product is already available then do not fetch the information again.